src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,148 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (addFirst(newNode)) {
+            addTail(newNode);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == 0) {
+            addFirstElement(value, newNode);
+        } else if (index == size) {
+            addTail(newNode);
```

Comment:
move this `if (index == size)` block on the first place (before `if (index ==0)`. in this way:
```
if (index == size) {
      add(value);
      return;
}
```
this help to simplify `if else `constructions and help to avoid additional check that list is not empty when we put new element to `index = 0`